behat/tests/bootstrap/v2/reporting/sections/ActionsSectionTrait.php

```diff
@@ -0,0 +1,143 @@
+            'first comment',
+            'action[doYouExpectFinancialDecisionsDetails]'
+        );
+    }
+
+    /**
+     * @Given I choose no and save on concerns actions section
+     */
+    public function iChooseNoOnDoYouHaveConcernsActionsSection()
+    {
+        $this->fillInActionsForm(
+            'no',
+            'action[doYouHaveConcerns]',
+        );
+    }
+
+    /**
+     * @Given I choose yes and save on concerns actions section
+     */
+    public function iChooseYesOnDoYouHaveConcernsActionsSection()
+    {
+        $this->fillInActionsForm(
+            'yes',
+            'action[doYouHaveConcerns]',
+            'second comment',
+            'action[doYouHaveConcernsDetails]'
+        );
+    }
+    public function fillInActionsForm($answer, $actionName, $comment = null, $commentName = null)
+    {
+        $this->selectOption($actionName, $answer);
+        if ($answer === "no") {
+            $this->answeredNo += 1;
+        } elseif ($answer === "yes") {
+            $this->answeredYes += 1;
+        }
+        if ($comment != null) {
+            $this->fillField($commentName, $comment);
+            array_push($this->comments, $comment);
+        }
+        $this->pressButton('Save and continue');
+    }
+
+    /**
+     * @When I should see the expected action report section responses
+     */
+    public function iSeeExpectedActionSectionResponses()
+    {
+        $table = $this->getSession()->getPage()->find('css', 'dl');
+        if (!$table) {
+            $this->throwContextualException('A dl element was not found on the page');
+        }
+        $tableEntry = $table->findAll('css', 'dd');
+        if (!$tableEntry) {
+            $this->throwContextualException('A dd element was not found on the page');
+        }
+        $countNegativeReponse = 0;
+        $countPositiveReponse = 0;
+        foreach ($tableEntry as $entry) {
+            if (trim(strtolower($entry->getHtml())) === "no") {
+                $countNegativeReponse += 1;
+            } elseif (strtolower(trim($entry->getHtml())) === "yes") {
+                $countPositiveReponse += 1;
+            }
+        }
+        assert($countNegativeReponse == $this->answeredNo);
+        assert($countPositiveReponse == $this->answeredYes);
+    }
+
+    /**
+     * @Then I should see the expected action comments
+     */
+    public function iShouldSeeTheExpectedActionComments()
+    {
+        $table = $this->getSession()->getPage()->find('css', 'dl');
+        if (!$table) {
+            $this->throwContextualException('A dl element was not found on the page');
+        }
+
+        foreach ($this->comments as $comment) {
+            assert(str_contains($table->getHtml(), $comment));
+        }
+    }
+
+    /**
+     * @Then I follow edit link on concerns question
+     */
+    public function iFollowEditLinkConcernActionsPage()
+    {
+        //this should be replaced with actual link click but could not identify it properly
```

Comment:
I'm assuming there was no id, unique class or label we could hook in to here? Something worth discussing at a higher level to see if we can make a change to always apply an id to links based on the element (or something along those lines)

Reply:
yeah nothing specific. the attribute of the href.. but then that would still be doing a visitpath instead of a clicklink. So yeah maybe we should apply an id as you say to these.